flang/lib/Lower/Bridge.cpp

```diff
@@ -1596,7 +1594,7 @@ class FirConverter : public Fortran::lower::AbstractConverter {
       Fortran::lower::pft::Evaluation *successor{};
       if (eval.isActionStmt())
         successor = eval.controlSuccessor;
-      else if (eval.isConstruct() && !eval.evaluationList->empty() &&
+      else if (eval.isConstruct() &&
```

Comment:
Can we at least have an assertion here? It's kind of a pain to have the code crash 20 or so frames later (or worse not at all) when it dereferences garbage pointing to protected memory.